strawberryfields/program.py

```diff
@@ -464,6 +469,49 @@ def assert_number_of_modes(self, device):
                 f"only supports a {device.modes}-mode program."
             )
 
+    def assert_number_of_measurements(self, device):
+        """Check that the number of measurements in the circuit doesn't exceed the number of allowed
+        measurements according to the device specification.
+
+        Args:
+            device (~strawberryfields.api.DeviceSpec): Device specification object to use.
+                ``device.modes`` must be a dictionary, containing the maximum number of allowed
+                measurements for the specified target.
+
+        """
+        num_pnr, num_homodyne, num_heterodyne = 0, 0, 0
+
+        try:
+            max_pnr = device.modes["max"]["pnr"]
+            max_homodyne = device.modes["max"]["homodyne"]
+            max_heterodyne = device.modes["max"]["heterodyne"]
+        except (KeyError, TypeError) as e:
+            raise KeyError(
+                "Device specification must contain an entry for the maximum allowed number "
+                "of measurments. Have you specified the correct target?"
+            ) from e
+
+        for c in self.circuit:
+            op_name = str(c.op)
+            if "MeasureFock" in op_name:
+                num_pnr += len(c.reg)
+            if "MeasureHomodyne" in op_name or "MeasureX" in op_name or "MeasureP" in op_name:
```

Comment:
`elif` could be useful, the rest is minor suggestion
```suggestion
            elif any(homodyne in op_name for homodyne in ("MeasureHomodyne", "MeasureX", "MeasureP")):
```

Reply:
Oh, true, I had `if`s instead of `elif`s there since I was thinking that there could be several type of measurements in a circuit, but this only checks one statement at a time, which will only contain a single measurement. 🤦  Thanks for catching this!